### PR TITLE
Change shrink script to use args instead of fstab

### PR DESCRIPTION
**Pull request by ygalblum**
Update the script to get the list of devices and size as a parameter
Change the shrink-start to a jinja template to inject the parameters
Update the tasks list and parameters to accommodate the change
Update the role README file

### CODE REVIEW

**Comment by ygalblum** [comment]
@jordigilh @swapdisk PTAL

File: infra.lvm_tools/shrink_lv/roles/shrink_lv/files/shrink.sh [hunk 1]
```diff
@@ -185,36 +141,24 @@ function process_entry() {
     return $?
 }
 
-
 function display_help() {
-    echo "Program to shrink an ext4 file system hosted in a Logical Volume. It retrieves the new size from the value of the option named systemd.shrinkfs captured in the /etc/fstab entry of the device. 
-    
+    echo "Program to shrink an ext4 file system hosted in a Logical Volume. It retrieves the new size from the value of the option named systemd.shrinkfs captured in the /etc/fstab entry of the device.
+
     Usage: '$(basename "$0")' [-h] [-d=|--device=|--all]
```
Comment (jordigilh) [inline infra.lvm_tools/shrink_lv/roles/shrink_lv/files/shrink.sh]: Shouldn't the help message be updated to reflect the new format as well? `--device=<device_name>:<new_size>`
https://github.com/ygalblum/ansible-bash-shrink-lv/blob/ebc04e0c648f43539627b7ba8fbbb841a7c12999/infra.lvm_tools/shrink_lv/roles/shrink_lv/templates/shrink-start.sh.j2#L19

**Comment by jordigilh** [comment]
@ygalblum I wonder if the PR's code structure is correct, because I'm seeing `shrink_lv` used 2 times in the path for the `shrink.sh` script.
`infra.lvm_tools/shrink_lv/roles/shrink_lv/files/shrink.sh`

Shouldn't it be more like this?
`infra.lvm_tools/roles/shrink_lv/files/shrink.sh`

**Comment by jordigilh** [comment]
Tested with success on my VM RHEL7 instance by shrinking the home LV, which is not active in the pre-mount stage. I only have minor cosmetic comments but the changes look good! 😄

**Comment by ygalblum** [comment]
> @ygalblum I wonder if the PR's code structure is correct, because I'm seeing `shrink_lv` used 2 times in the path for the `shrink.sh` script. `infra.lvm_tools/shrink_lv/roles/shrink_lv/files/shrink.sh`
> 
> Shouldn't it be more like this? `infra.lvm_tools/roles/shrink_lv/files/shrink.sh`

Not sure what you mean by 2 times.

The directory structure created by `ansible-galaxy collection init` is `<Namespace>/<Collection>` under which you can find the `roles` directory. In this case, the Namespace was set to `infra.lvm_tools` and the Collection to `shrink_lv`.

I agree that this might not be the structure we would have liked to see. But, we decided that once these fixes are merged, the role should be merged into the first collection. So, no point in fixing the structure here.

File: infra.lvm_tools/shrink_lv/roles/shrink_lv/files/shrink.sh (hunk 1 above)
Comment (ygalblum) [reply infra.lvm_tools/shrink_lv/roles/shrink_lv/files/shrink.sh]: I've updated the help message

**Pull request merged**